homeassistant/components/nextdns/__init__.py

```diff
@@ -132,14 +135,27 @@ async def _async_update_data(self) -> AnalyticsProtocols:
             raise UpdateFailed(err) from err
 
 
+class NextDnsSettingsUpdateCoordinator(NextDnsUpdateCoordinator[Settings]):
+    """Class to manage fetching NextDNS connection data from API."""
+
+    async def _async_update_data(self) -> Settings:
+        """Update data via library."""
+        try:
+            async with timeout(10):
+                return await self.nextdns.get_settings(self.profile_id)
+        except (ApiError, ClientConnectorError, InvalidApiKeyError) as err:
+            raise UpdateFailed(err) from err
+
+
```

Comment:
This code seems to be duplicated in all coordinators.
Maybe the try...except could be moved to the base class, and in the child classes only have
```suggestion
    async def _async_update_data_internal(self) -> Settings:
        """Update data via library."""
        return await self.nextdns.get_settings(self.profile_id)


```

Reply:
Good idea!